EDU-1043/1048: get Cloud Account Id and generate-certificates commands (#2297)
temporalio/documentation: docs-src/cloud/tcld/generate-certificates/certificate-authority-certificate.md

---
id: certificate-authority-certificate
title: tcld generate-certificates certificate-authority-certificate
sidebar_label: certificate-authority-certificate
description: Generate CA certificates using tcld.
tags:
  - tcld
  - cli-reference
---

The `tcld generate-certificates certificate-authority-certificate` command generates certificate authority (CA) certificates for Temporal Cloud.

`tcld generate-certificates certificate-authority-certificate <modifiers>`

Alias: `ca`

The following modifiers control the behavior of the command.

#### --organization

Specify an organization name for certificate generation.

Alias: `--org`

**Example**

```bash
tcld generate-certificates certificate-authority-certificate --organization <value>
```

#### --validity-period

> Specify the duration for which the certificate is valid for.
> Format values as d/h (e.g. 30d10h for a certificate lasting 30 days and 10 hours.)

Reviewer: ```suggestion
Specify the duration for which the certificate is valid.
Format values as d/h (for example, `30d10h` for a certificate lasting 30 days and 10 hours).
```
Perhaps change "d/h" to `#d#h`? Or something similar?

Alias: `-d`

**Example**

```bash
tcld generate-certificates certificate-authority-certificate --validity-period <value>
```

#### --ca-certificate-file

> Specify a path where the generated x509 certificate file will be stored.

Reviewer: ```suggestion
Specify a path where the generated X.509 certificate file will be stored.
```

Alias: `--ca-cert`

**Example**

```bash
tcld generate-certificates certificate-authority-certificate --ca-certificate-file <path>
```

#### --ca-key-file

Specify a path where the certificate's private key will be stored.

Alias: `--ca-key`

**Example**

```bash
tcld generate-certificates certificate-authority-certificate --ca-key-file <path>
```

#### --rsa-algorithm

> When enabled, a 4096-bit RSA keypair is generated for the certificate instead of an ECDSA P-384 keypair.
> Since an ECDSA P-384 keypair is the recommended default, this option is disabled.

Reviewer: ```suggestion
When enabled, a 4096-bit RSA key pair is generated for the certificate instead of an ECDSA P-384 key pair.
Because an ECDSA P-384 key pair is the recommended default, this option is disabled.
```
Per https://developers.google.com/style/word-list#key-pair.

Alias: `--rsa`

> Specify a path to store the generated x509 certificate file.

Reviewer: ```suggestion
Specify a path to store the generated X.509 certificate file.
```
